quinn/src/endpoint.rs

```diff
@@ -137,8 +138,10 @@ impl Endpoint {
 
     /// Get the next incoming connection attempt from a client
     ///
-    /// Yields [`Connecting`] futures that must be `await`ed to obtain the final `Connection`, or
-    /// `None` if the endpoint is [`close`](Self::close)d.
+    /// Yields [`Incoming`]s, or `None` if the endpoint is [`close`](Self::close)d. [`Incoming`]
```

Comment:
Connecting to my comment on the IncomingFuture adapter not really having an obvious use - perhaps a more useful  convenience utility would be to be able to configure an Endpoint to accept, reject or verify with a supplied predicate. Something like 

```
enum FilterDecision {
    Accept,
    Reject,
    Validate,
}

filter_incoming: F where F: FnMut(incoming: Incoming, remote: IpAddr) -> FilterDecision
```

Reply:
That seems strictly less powerful, and also more difficult to use, than inserting whatever logic you like after a successful `Endpoint::accept`.

Reply:
I think that sort of "accept/reject/retry logic as middleware" architecture would be good for a high level framework, but for a low-level transport implementation we should tend towards giving the user more control.